shell/common/shell_test_platform_view_metal.mm

```diff
@@ -8,64 +8,50 @@
 
 #include <utility>
 
-#include "flutter/fml/platform/darwin/scoped_nsobject.h"
 #include "flutter/shell/gpu/gpu_surface_metal_impeller.h"
 #include "flutter/shell/gpu/gpu_surface_metal_skia.h"
 #include "flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalImpeller.h"
 #include "flutter/shell/platform/darwin/graphics/FlutterDarwinContextMetalSkia.h"
 
+FLUTTER_ASSERT_ARC
+
 namespace flutter {
 namespace testing {
 
-static fml::scoped_nsprotocol<id<MTLTexture>> CreateOffscreenTexture(id<MTLDevice> device) {
+// This is out of the header so that shell_test_platform_view_metal.h can be included in
+// non-Objective-C TUs.
+struct DarwinContextMetal {
+  FlutterDarwinContextMetalSkia* skia_context;
+  FlutterDarwinContextMetalImpeller* impeller_context;
+  id<MTLTexture> offscreen_texture;
+};
+
+static std::unique_ptr<DarwinContextMetal> CreateDarwinContext(
+    bool impeller,
+    const std::shared_ptr<const fml::SyncSwitch>& is_gpu_disabled_sync_switch) {
+  FlutterDarwinContextMetalSkia* skia_context = nil;
+  FlutterDarwinContextMetalImpeller* impeller_context = nil;
+  id<MTLDevice> device = nil;
+  if (impeller) {
+    impeller_context = [[FlutterDarwinContextMetalImpeller alloc] init:is_gpu_disabled_sync_switch];
+    FML_CHECK(impeller_context.context);
+    device = impeller_context.context->GetMTLDevice();
+  } else {
+    skia_context = [[FlutterDarwinContextMetalSkia alloc] initWithDefaultMTLDevice];
+    FML_CHECK(skia_context.mainContext);
+    device = skia_context.device;
+  }
   auto descriptor =
       [MTLTextureDescriptor texture2DDescriptorWithPixelFormat:MTLPixelFormatBGRA8Unorm
                                                          width:800
                                                         height:600
                                                      mipmapped:NO];
   descriptor.usage = MTLTextureUsageRenderTarget | MTLTextureUsageShaderRead;
-  return fml::scoped_nsprotocol<id<MTLTexture>>{[device newTextureWithDescriptor:descriptor]};
+  id<MTLTexture> offscreen_texture = [device newTextureWithDescriptor:descriptor];
+  return std::make_unique<DarwinContextMetal>(
+      DarwinContextMetal{skia_context, impeller_context, offscreen_texture});
 }
 
-// This is out of the header so that shell_test_platform_view_metal.h can be included in
-// non-Objective-C TUs.
-class DarwinContextMetal {
- public:
-  explicit DarwinContextMetal(
-      bool impeller,
-      const std::shared_ptr<const fml::SyncSwitch>& is_gpu_disabled_sync_switch)
-      : context_(impeller ? nil : [[FlutterDarwinContextMetalSkia alloc] initWithDefaultMTLDevice]),
-        impeller_context_(
-            impeller ? [[FlutterDarwinContextMetalImpeller alloc] init:is_gpu_disabled_sync_switch]
-                     : nil),
-        offscreen_texture_(CreateOffscreenTexture(
-            impeller ? [impeller_context_ context]->GetMTLDevice() : [context_ device])) {}
```

Comment:
This is now done in the static `CreateDarwinContext` func above, which inlines `CreateOffscreenTexture` since it was only used here.

None of the rest of the getter boilerplate is necessary.